Popular_Models_Guide/Llama2/trtllm_guide.md

```diff
@@ -0,0 +1,129 @@
+
+## Installation
+
+Launch Triton docker container with TensorRT-LLM backend
+```docker run --rm -it --net host --shm-size=2g --ulimit memlock=-1 --ulimit stack=67108864 --gpus all -v /path/to/tensorrtllm_backend:/tensorrtllm_backend nvcr.io/nvidia/tritonserver:23.10-trtllm-py3 bash```
+
+Alternatively, you can follow instructions [here](https://github.com/triton-inference-server/tensorrtllm_backend/blob/main/README.md) to build Tritonserver with Tensorrt-LLM Backend if you want to build a specialized container.
+
+Don't forget to allow gpu usage when you launch the container.
+
+## Create Engines for each model [skip this step if you already have an engine]
+TensorRT-LLM requires each model to be compiled for the configuration you need before running.
+To do so, before you run your model for the first time on Tritonserver you will need to create a TensorRT-LLM engine for the model for the configuration you want.
+To do so, you will need to complete the following steps:
+
+1. Install Tensorrt-LLM python package
+   ```bash
+    # TensorRT-LLM is required for generating engines.
+    pip install git+https://github.com/NVIDIA/TensorRT-LLM.git
+    mkdir /usr/local/lib/python3.10/dist-packages/tensorrt_llm/libs/
+    cp /opt/tritonserver/backends/tensorrtllm/* /usr/local/lib/python3.10/dist-packages/tensorrt_llm/libs/
+    ```
+
+2.  Log in to huggingface-cli
+
+    ```bash
+    huggingface-cli login --token hf_*****
+    ```
+
+3.  Compile model engines
+    The script to build Llama models is located in [TensorRT-LLM repository](https://github.com/NVIDIA/TensorRT-LLM/tree/main/examples). We use the one located in the docker container as
+     `/tensorrtllm_backend/tensorrt_llm/examples/llama/build.py`.
+     This command compiles the model with in flight batching and 1 GPU. More details for the scripting please see the documentation for the Llama example [here](https://github.com/NVIDIA/TensorRT-LLM/tree/main/examples/llama/README.md).
+
+    ```bash
+    python build.py --model_dir /<path to your llama repo>/Llama-2-7b-chat-hf/ \
+                    --dtype bfloat16 \
+                    --use_gpt_attention_plugin bfloat16 \
+                    --use_inflight_batching \
+                    --paged_kv_cache \
+                    --remove_input_padding \
+                    --use_gemm_plugin bfloat16 \
+                    --output_dir /<path to your engine>/1-gpu/
+                    --world-size 1
+    ```
+
+    > Optional: You can check test the output of the model with `run.py`
+    > located in the same llama examples folder.
+    >
+    >   ```bash
+    >    python3 /run.py --engine_dir=<path to your engine>/1-gpu/ --max_output_len 100 --tokenizer_dir <path to your llama repo>/Llama-2-7b-chat-hf --input_text "How do I count to ten in French?"
+    >    ```
+
+## Serving with Triton
+
+We're almost there! The last step is to create a Triton readable model. You can
```

Comment:
(Optional) To me "We're almost there!" reads as if the tutorial up to this point has taken an unreasonable amount of time. It may dissuade someone who is reading and not following the tutorial from giving it a try. But I may also be in my own head about this...your call :)

Reply:
It was mostly me complaining of how long it is. But it's a lot simpler now so I will remove this line